src/frontend/src/lib/derived/exchange.derived.ts

```diff
@@ -24,9 +24,9 @@ export const exchanges: Readable<ExchangesData> = derived(
 			[SEPOLIA_TOKEN_ID]: ethPrice,
 			[ICP_TOKEN_ID]: icpPrice,
 			...Object.entries($exchangeStore ?? {}).reduce((acc, [key, currentPrice]) => {
-				const token = $erc20Tokens.find(
-					({ address }) => address.toLowerCase() === key.toLowerCase()
-				);
+				const token =
+					$erc20Tokens.find(({ address }) => address.toLowerCase() === key.toLowerCase()) ??
```

Comment:
Does that work? Arent' the Icrc token IDs added twice to the object?

```
....
// your code
const token = $erc20To ?? $icrcTokens.find(({ ledgerCanisterId }) => ledgerCanisterId === key);

// later in the same module
...$icrcTokens.reduce((acc, token) => {
				const { id, exchangeCoinId } = token;
				
				etc.
```

Reply:
From what I understood in the code, that part is done to set the prices of ckBTC and ckETH tokens (that otherwise we wouldn't have before), but even, for some reason that I am not sure about, to have a price in case of custom tokens (see function `loadMetadata`).

I didn't want to change that, BUT you are right, I was confused: if the deconstructed array comes after it will overwrite the values, and not the contrary. I will invert the position of those two parts of the return